commit ca89c49fa2c5cbb3945897046f33eed9f7da846c
Author: Jiang Jiang <jiang.jiang@nokia.com>
Date:   Tue Jul 3 10:17:49 2012 +0200

    Fix cursor truncate to include line position
    
    Since we could have moved the line position (QTextLine::setPosition),
    the truncating position should be adjusted with that.
    
    Change-Id: Ie1acd4a1b6a4adfbeeb9ce8ee2dfa19d992470e8
    Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@nokia.com>

diff --git a/src/gui/text/qtextlayout.cpp b/src/gui/text/qtextlayout.cpp
index 16f7150..52f2793 100644
--- a/src/gui/text/qtextlayout.cpp
+++ b/src/gui/text/qtextlayout.cpp
@@ -2616,8 +2616,8 @@ qreal QTextLine::cursorToX(int *cursorPos, Edge edge) const
         x += eng->offsetInLigature(si, pos, end, glyph_pos);
     }
 
-    if (eng->option.wrapMode() != QTextOption::NoWrap && x > line.width)
-        x = line.width;
+    if (eng->option.wrapMode() != QTextOption::NoWrap && x > line.x + line.width)
+        x = line.x + line.width;
 
     *cursorPos = pos + si->position;
     return x.toReal();
